src/human-readable/runtime/utils.ts

```diff
@@ -0,0 +1,225 @@
+  execErrorSignature,
+  execEventSignature,
+  execFunctionSignature,
+  isConstructorSignature,
+  isErrorSignature,
+  isEventSignature,
+  isFallbackSignature,
+  isFunctionSignature,
+  isReceiveSignature,
+} from './signatures'
+
+export function parseSignature(signature: string, structs: StructLookup = {}) {
+  if (isFunctionSignature(signature)) {
+    const match = execFunctionSignature(signature)
+    if (!match)
+      throw new BaseError('Invalid function signature.', {
+        details: signature,
+      })
+    const inputParams = splitParameters(match.parameters)
+    const inputs = []
+    const inputLength = inputParams.length
+    for (let i = 0; i < inputLength; i++) {
+      inputs.push(
+        parseAbiParameter(inputParams[i]!, {
+          structs,
+          modifiers: ['calldata', 'memory', 'storage'],
+        }),
+      )
+    }
+
+    const outputs = []
+    if (match.returns) {
+      const outputParams = splitParameters(match.returns)
+      const outputLength = outputParams.length
+      for (let i = 0; i < outputLength; i++) {
+        outputs.push(parseAbiParameter(outputParams[i]!, { structs }))
+      }
+    }
+
+    return {
+      name: match.name,
+      type: 'function',
+      stateMutability: match.stateMutability ?? 'nonpayable',
+      inputs,
+      outputs,
+    }
+  }
+
+  if (isEventSignature(signature)) {
+    const match = execEventSignature(signature)
+    if (!match)
+      throw new BaseError('Invalid event signature.', {
+        details: signature,
+      })
+    const params = splitParameters(match.parameters)
+    const abiParameters = []
+    const length = params.length
+    for (let i = 0; i < length; i++) {
+      abiParameters.push(
+        parseAbiParameter(params[i]!, {
+          structs,
+          modifiers: ['indexed'],
+          type: 'event',
+        }),
+      )
+    }
+    return { name: match.name, type: 'event', inputs: abiParameters }
+  }
+
+  if (isErrorSignature(signature)) {
+    const match = execErrorSignature(signature)
+    if (!match)
+      throw new BaseError('Invalid error signature.', {
+        details: signature,
+      })
+    const params = splitParameters(match.parameters)
+    const abiParameters = []
+    const length = params.length
+    for (let i = 0; i < length; i++) {
+      abiParameters.push(parseAbiParameter(params[i]!, { structs }))
+    }
+    return { name: match.name, type: 'error', inputs: abiParameters }
+  }
+
+  if (isConstructorSignature(signature)) {
+    const match = execConstructorSignature(signature)
+    if (!match)
+      throw new BaseError('Invalid constructor signature.', {
+        details: signature,
+      })
+    const params = splitParameters(match.parameters)
+    const abiParameters = []
+    const length = params.length
+    for (let i = 0; i < length; i++) {
+      abiParameters.push(parseAbiParameter(params[i]!, { structs }))
+    }
+    return {
+      type: 'constructor',
+      stateMutability: 'nonpayable',
```

Comment:
here it could possibly be `stateMutability: match.stateMutability ?? 'nonpayable'`